libyul/AsmPrinter.cpp

```diff
@@ -266,7 +272,7 @@ string AsmPrinter::formatTypedName(TypedName _variable) const
 
 string AsmPrinter::appendTypeName(YulString _type) const
 {
-	if (m_yul)
+	if (m_yul && !_type.str().empty())
```

Comment:
Unrelated feature?

Reply:
No. This is a seperate commit. And I needed it in order to print the Yul AST when no type information is there. Maybe I should have set `m_yul` to `false`, but I ended up having all values with a trailing colon but no type attached, as the source code didn't contain types. I am open for suggestions here, though.

Reply:
Can you create a new PR just for that?

Reply:
for reference: #6883